### PR TITLE
[candidate_parameters] Update TestPlan to include DoB and DoD tabs

### LINKED ISSUE #6213: Add testplan for DOB and DOD tabs of candidate params
**Describe the bug**
Missing test plan sections for new feature

### PR DESCRIPTION
**Pull request by jesscall**
## Brief summary of changes
Adding instructions to TestPlan for the following new features:
* Date of Birth Tab (#4915)
* Date of Death Tab (#4929)

#### Resolves #6213

Assigning @maltheism since she is responsible for testing candidate parameters during this testing round.

### CODE REVIEW

File: modules/candidate_parameters/test/TestPlan.md [hunk 1]
```diff
@@ -67,4 +67,18 @@
         * Error: date cannot be later than today
     * Consent to Study = No (changing/updating an already existing 'Yes' consent); Date (Withdrawal) of Consent = valid random date; Confirmation (Withdrawal) Date of Consent = same random date
         * No error
-        * Make sure they update properly in the front-end and backend
\ No newline at end of file
+        * Make sure they update properly in the front-end and backend
+
+### Date of Birth Tab
+32. Check that date of birth can only be amended if user has candidate_dob_edit permission.
+33. Change the date of birth and try to save it. Check that it is updated on front-end and back-end.
```
Comment (ridz1208) [inline modules/candidate_parameters/test/TestPlan.md:74]:
```suggestion
33. Change the date of birth and try to save it. Check that it is updated on front-end and database.
```

File: modules/candidate_parameters/test/TestPlan.md [hunk 2]
```diff
@@ -67,4 +67,18 @@
...
+34. Check that entering a date later than today gives a data entry error.
+35. Check that the date stored matches the _DOB Format_ specified in the configuration module. 
+
+### Date of Death Tab
+36. Check that date of death can only be set if user has candidate_dod_edit permission.
+37. Change the date of death and try to save it. Check that it is updated on front-end and back-end.
```
Comment (ridz1208) [inline modules/candidate_parameters/test/TestPlan.md:80]:
```suggestion
37. Change the date of death and try to save it. Check that it is updated on front-end and database.
```

File: modules/candidate_parameters/test/TestPlan.md [hunk 3]
```diff
@@ -67,4 +67,18 @@
...
+38. Check that entering a date of death before the date of birth gives a data entry error.
+39. Check that entring a date later than today gives a date entry error.
```
Comment (ridz1208) [inline modules/candidate_parameters/test/TestPlan.md:82]:
```suggestion
39. Check that entering a date later than today gives a data entry error.
```

File: modules/candidate_parameters/test/TestPlan.md [hunk 4]
```diff
@@ -67,4 +67,18 @@
...
+40. Check that the date stored matches the _DOD Format_ specified in the configuration module.
+41. Check that filling out any instrument with a _Date of Administration_ later than the DoD changes the age displayed to the candidate's age at death (in months).
```
Comment (ridz1208) [inline modules/candidate_parameters/test/TestPlan.md:84]: this doesn't seem to belong here, it should be in the `instruments` test_plan which doesn't exist yet.

@PapillonMcGill when you start working on https://github.com/aces/Loris/issues/6293. keep that in mind

File: modules/candidate_parameters/test/TestPlan.md [hunk 5]
```diff
@@ -67,4 +67,18 @@
...
+        * Make sure they update properly in the front-end and backend
+
+### Date of Birth Tab
+32. Check that date of birth can only be amended if user has candidate_dob_edit permission.
+33. Change the date of birth and try to save it. Check that it is updated on front-end and back-end.
+34. Check that entering a date later than today gives a data entry error.
+35. Check that the date stored matches the _DOB Format_ specified in the configuration module. 
+
+### Date of Death Tab
+36. Check that date of death can only be set if user has candidate_dod_edit permission.
```
Comment (ridz1208) [inline modules/candidate_parameters/test/TestPlan.md:79]: does the tab show to the user if they dont have the permission ?

File: modules/candidate_parameters/test/TestPlan.md [hunk 6]
```diff
@@ -67,4 +67,18 @@
         * Error: date cannot be later than today
     * Consent to Study = No (changing/updating an already existing 'Yes' consent); Date (Withdrawal) of Consent = valid random date; Confirmation (Withdrawal) Date of Consent = same random date
         * No error
-        * Make sure they update properly in the front-end and backend
\ No newline at end of file
+        * Make sure they update properly in the front-end and backend
+
+### Date of Birth Tab
+32. Check that date of birth can only be amended if user has candidate_dob_edit permission.
```
Comment (ridz1208) [inline modules/candidate_parameters/test/TestPlan.md:73]: does the tab show to the user if they dont have the permission ?

File: modules/candidate_parameters/test/TestPlan.md (hunk 1 above)
Comment (maltheism) [reply modules/candidate_parameters/test/TestPlan.md:74]: maybe `back-end database` would be better?

File: modules/candidate_parameters/test/TestPlan.md (hunk 5 above)
Comment (jesscall) [reply modules/candidate_parameters/test/TestPlan.md:79]: yes, however the field is disabled.

File: modules/candidate_parameters/test/TestPlan.md (hunk 6 above)
Comment (jesscall) [reply modules/candidate_parameters/test/TestPlan.md:73]: yes, however the field is disabled.

**Comment by ridz1208** [comment]
@jesscall keeping the changes requested for item 41 to be removed

**Comment by jesscall** [comment]
@ridz1208 resolved!

**Comment by ridz1208** [comment]
@jesscall your PR was oncorrectly based on master
its already approved but can you please rebase it on 23 so it goes in the latest release ?

**Comment by jesscall** [comment]
@ridz1208 done.

**Comment by ridz1208** [comment]
@driusan all urs

**Pull request merged**